samcli/commands/local/lib/sam_api_provider.py

```diff
@@ -280,6 +321,7 @@ def _convert_event_api(lambda_logical_id, event_properties):
         """
         path = event_properties.get(SamApiProvider._EVENT_PATH)
         method = event_properties.get(SamApiProvider._EVENT_METHOD)
+        stage_name = event_properties.get("StageName")
```

Comment:
Here, as far as I know in the SAM spec, you should get ```None``` as the StageName property.

```None``` is a placeholder: if the event references an explicit API with a set StageName property, ```None``` gets changed to the given stage name. If it is a true implicit (no associated API property, just SAM generated API) API, then the StageName will stay as ```None``` and will be changed to the SAM default API stage name later.